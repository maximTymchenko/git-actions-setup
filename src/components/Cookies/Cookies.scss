// import vars
@import '../../styles/vars';

.coockies {
    &__wrapper {
        width: 100%;
        background-color: #fff;
        z-index: 99;
        position: fixed;
        bottom: 7px;
        left: 0;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border: 2px solid $secondary;
        box-shadow: 5px 5px 1px 1px rgba(57, 156, 203, 0.4);

        &:active {
            box-shadow: 5px 5px 1px 1px rgba(57, 156, 203, 0);
        }

        @media screen and (max-width: $mobile) {
            padding: 15px 0 20px;
            display: flex;
            align-items: center;
            flex-direction: column;
        }
    }
    &__info {
        max-width: 800px;
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        color: $text__body;
        padding: 13px 50px;

        @media screen and (max-width: $mobile) {
            margin-bottom: 15px;
        }
    }
    &__privacy {
        cursor: pointer;
        font-family: 'OktaNeue-Medium';
        font-size: 15px;
        font-weight: 500;
        line-height: 1.53;
        color: $secondary;
    }

    &__btn {
        background: none;
        outline: none;
        border: none;
        font-family: 'OktaNeue-Black';
        cursor: pointer;
        padding-right: 50px;
        font-size: 15px;
        font-weight: 900;
        letter-spacing: 0.5px;
        color: $primary__light;
    }

    &__link {
        text-decoration: none;
    }
}
