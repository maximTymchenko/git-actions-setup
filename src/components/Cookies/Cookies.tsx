import React, { useState } from 'react';
import { Link } from 'gatsby';
import './Cookies.scss';

const Cookies: React.FunctionComponent = () => {
    const getItem = (name: string) => {
        if (typeof document !== `undefined`) {
            const cookies: any = document.cookie
                .split(';')
                .map((cookie: string) => cookie.split('='))
                .reduce((acc, [key, value]) => ({ ...acc, [`${key.trim()}`]: value }), {});
            return cookies[name];
        }
    };

    const setItem = (key: string, value: string): void => {
        // set cookies expired time
        const cookiesExpire = `${key}=${value} ;max-age =` + 60 * 60 * 24 * 90;
        document.cookie = cookiesExpire;
    };

    const acceptCookiesKey: string = 'accept_cookies';

    //manage cookies state
    const [showCookies, setShowCookies] = useState<boolean>(getItem(acceptCookiesKey) != 'true');

    const acceptFn = (): void => {
        setItem(acceptCookiesKey, 'true');
        setShowCookies(false);
    };

    return (
        <div data-testid="cookies" className="cookies">
            {showCookies && (
                <div className="coockies__wrapper">
                    <p className="coockies__info">
                        We use cookies to ensure your best experience. Through your continued use of
                        this site you accept this use. Please see our
                        <Link className="coockies__link" to="/privacy-policy">
                            <span className="coockies__privacy"> privacy policy</span>
                        </Link>
                    </p>
                    <button onClick={() => acceptFn()} className="coockies__btn">
                        Accept
                    </button>
                </div>
            )}
        </div>
    );
};

export default Cookies;
