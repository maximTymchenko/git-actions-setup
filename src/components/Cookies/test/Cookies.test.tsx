import React from 'react'
import Cookies from '../Cookies'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(<Cookies></Cookies>)
  expect(getByTestId('cookies')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer.create(<Cookies></Cookies>).toJSON()
  expect(tree).toMatchSnapshot()
})
