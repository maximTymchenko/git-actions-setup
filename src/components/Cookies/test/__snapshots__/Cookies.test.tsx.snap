// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="cookies"
  data-testid="cookies"
>
  <div
    className="coockies__wrapper"
  >
    <p
      className="coockies__info"
    >
      We use cookies to ensure your best experience. Through your continued use of this site you accept this use. Please see our
      <a
        className="coockies__link"
        href="/privacy-policy"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span
          className="coockies__privacy"
        >
           privacy policy
        </span>
      </a>
    </p>
    <button
      className="coockies__btn"
      onClick={[Function]}
    >
      Accept
    </button>
  </div>
</div>
`;
